### DISCUSSION TITLE
[Question]Does recoil preserve immutability?

### DISCUSSION THREAD

**Discussion by monii** [facebookexperimental/Recoil#1733] [Q&A] [answered]
hello??
Currently, we are trying to manage the state by applying recoil to react.
I have one question. Does recoil preserve immutability when changing atoms?
I don't see many examples of using recoil and immer.js together, so I wonder if reference type values ​​are deep copy.
There was a question on stackoverflow that had a similar problem to me, but it didn't have an answer, so I'm posting it here!
[stackoverflow](https://stackoverflow.com/questions/69844772/are-react-recoil-state-objects-silently-immutable)

I'd really appreciate it if you could give me the answer! :)

**Comment by drarmstr** [comment, marked as answer]
Recoil does not deep copy state values.  However, state values should be immutable and Recoil will actually deep freeze them in dev mode to catch issues.  (Except with `dangerouslyAllowMutability` option)
